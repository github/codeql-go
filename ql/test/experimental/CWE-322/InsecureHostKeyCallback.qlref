experimental/CWE-322/InsecureHostKeyCallback.ql