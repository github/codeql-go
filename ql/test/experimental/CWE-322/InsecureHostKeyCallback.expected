edges
| InsecureHostKeyCallbackExample.go:13:4:15:3 | function literal : signature type | InsecureHostKeyCallbackExample.go:12:20:15:4 | type conversion |
| InsecureHostKeyCallbackExample.go:28:14:31:3 | type conversion : signature type | InsecureHostKeyCallbackExample.go:36:20:36:27 | callback |
| InsecureHostKeyCallbackExample.go:29:3:31:2 | function literal : signature type | InsecureHostKeyCallbackExample.go:28:14:31:3 | type conversion : signature type |
| InsecureHostKeyCallbackExample.go:42:3:44:2 | function literal : signature type | InsecureHostKeyCallbackExample.go:49:20:49:48 | type conversion |
| InsecureHostKeyCallbackExample.go:53:39:53:46 | definition of callback : HostKeyCallback | InsecureHostKeyCallbackExample.go:57:20:57:27 | callback |
| InsecureHostKeyCallbackExample.go:53:39:53:46 | definition of callback : signature type | InsecureHostKeyCallbackExample.go:57:20:57:27 | callback |
| InsecureHostKeyCallbackExample.go:64:22:67:3 | type conversion : signature type | InsecureHostKeyCallbackExample.go:69:35:69:50 | insecureCallback : signature type |
| InsecureHostKeyCallbackExample.go:65:3:67:2 | function literal : signature type | InsecureHostKeyCallbackExample.go:64:22:67:3 | type conversion : signature type |
| InsecureHostKeyCallbackExample.go:69:35:69:50 | insecureCallback : signature type | InsecureHostKeyCallbackExample.go:53:39:53:46 | definition of callback : signature type |
| InsecureHostKeyCallbackExample.go:80:35:80:61 | call to InsecureIgnoreHostKey : HostKeyCallback | InsecureHostKeyCallbackExample.go:53:39:53:46 | definition of callback : HostKeyCallback |
nodes
| InsecureHostKeyCallbackExample.go:12:20:15:4 | type conversion | semmle.label | type conversion |
| InsecureHostKeyCallbackExample.go:13:4:15:3 | function literal : signature type | semmle.label | function literal : signature type |
| InsecureHostKeyCallbackExample.go:23:20:23:46 | call to InsecureIgnoreHostKey | semmle.label | call to InsecureIgnoreHostKey |
| InsecureHostKeyCallbackExample.go:28:14:31:3 | type conversion : signature type | semmle.label | type conversion : signature type |
| InsecureHostKeyCallbackExample.go:29:3:31:2 | function literal : signature type | semmle.label | function literal : signature type |
| InsecureHostKeyCallbackExample.go:36:20:36:27 | callback | semmle.label | callback |
| InsecureHostKeyCallbackExample.go:42:3:44:2 | function literal : signature type | semmle.label | function literal : signature type |
| InsecureHostKeyCallbackExample.go:49:20:49:48 | type conversion | semmle.label | type conversion |
| InsecureHostKeyCallbackExample.go:53:39:53:46 | definition of callback : HostKeyCallback | semmle.label | definition of callback : HostKeyCallback |
| InsecureHostKeyCallbackExample.go:53:39:53:46 | definition of callback : signature type | semmle.label | definition of callback : signature type |
| InsecureHostKeyCallbackExample.go:57:20:57:27 | callback | semmle.label | callback |
| InsecureHostKeyCallbackExample.go:64:22:67:3 | type conversion : signature type | semmle.label | type conversion : signature type |
| InsecureHostKeyCallbackExample.go:65:3:67:2 | function literal : signature type | semmle.label | function literal : signature type |
| InsecureHostKeyCallbackExample.go:69:35:69:50 | insecureCallback : signature type | semmle.label | insecureCallback : signature type |
| InsecureHostKeyCallbackExample.go:80:35:80:61 | call to InsecureIgnoreHostKey : HostKeyCallback | semmle.label | call to InsecureIgnoreHostKey : HostKeyCallback |
#select
| InsecureHostKeyCallbackExample.go:12:20:15:4 | type conversion | InsecureHostKeyCallbackExample.go:13:4:15:3 | function literal : signature type | InsecureHostKeyCallbackExample.go:12:20:15:4 | type conversion | Configuring SSH ClientConfig with insecure HostKeyCallback implementation from $@. | InsecureHostKeyCallbackExample.go:13:4:15:3 | function literal | this source |
| InsecureHostKeyCallbackExample.go:23:20:23:46 | call to InsecureIgnoreHostKey | InsecureHostKeyCallbackExample.go:23:20:23:46 | call to InsecureIgnoreHostKey | InsecureHostKeyCallbackExample.go:23:20:23:46 | call to InsecureIgnoreHostKey | Configuring SSH ClientConfig with insecure HostKeyCallback implementation from $@. | InsecureHostKeyCallbackExample.go:23:20:23:46 | call to InsecureIgnoreHostKey | this source |
| InsecureHostKeyCallbackExample.go:36:20:36:27 | callback | InsecureHostKeyCallbackExample.go:29:3:31:2 | function literal : signature type | InsecureHostKeyCallbackExample.go:36:20:36:27 | callback | Configuring SSH ClientConfig with insecure HostKeyCallback implementation from $@. | InsecureHostKeyCallbackExample.go:29:3:31:2 | function literal | this source |
| InsecureHostKeyCallbackExample.go:49:20:49:48 | type conversion | InsecureHostKeyCallbackExample.go:42:3:44:2 | function literal : signature type | InsecureHostKeyCallbackExample.go:49:20:49:48 | type conversion | Configuring SSH ClientConfig with insecure HostKeyCallback implementation from $@. | InsecureHostKeyCallbackExample.go:42:3:44:2 | function literal | this source |
| InsecureHostKeyCallbackExample.go:57:20:57:27 | callback | InsecureHostKeyCallbackExample.go:65:3:67:2 | function literal : signature type | InsecureHostKeyCallbackExample.go:57:20:57:27 | callback | Configuring SSH ClientConfig with insecure HostKeyCallback implementation from $@. | InsecureHostKeyCallbackExample.go:65:3:67:2 | function literal | this source |
| InsecureHostKeyCallbackExample.go:57:20:57:27 | callback | InsecureHostKeyCallbackExample.go:80:35:80:61 | call to InsecureIgnoreHostKey : HostKeyCallback | InsecureHostKeyCallbackExample.go:57:20:57:27 | callback | Configuring SSH ClientConfig with insecure HostKeyCallback implementation from $@. | InsecureHostKeyCallbackExample.go:80:35:80:61 | call to InsecureIgnoreHostKey | this source |
